lib/everypolitician/popolo/membership.rb

```diff
@@ -6,6 +6,11 @@ class Membership < Entity
       def person
         popolo.persons.find_by(id: person_id)
       end
+
+      def ==(other)
+        self.class == other.class && instance_variables.all? { |v| v == other[v] }
```

Comment:
I may be misunderstanding something subtle, but I don't think this is doing what you think it's doing. `v` here will be an instance variable name (e.g. `:@document`), so `other[v]`, I think is always going to be `nil` (not only off because of the syntax, but also the mismatch between testing the key of one against the value of the other)

If I'm understanding correctly what's going on, I think you'd need to be testing `self.instance_variable_get(v) == other.instance_variable_get(v)` (though @chrismytton may be able to suggest a way of simplifying that. My ruby metaprogramming isn't particularly strong.